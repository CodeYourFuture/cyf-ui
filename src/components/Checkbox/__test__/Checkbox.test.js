import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Checkbox } from "..";

describe("<Checkbox/>", () => {
  test("test matches snapshot", async () => {
    const mockOnChange = jest.fn(() => null);

    const { container } = render(<Checkbox checked onChange={mockOnChange} label="include VAT" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test("onChange handler is called when user clicked on the checkbox", async () => {
    const mockOnChange = jest.fn();

    const { container } = render(<Checkbox checked onChange={mockOnChange} label="include VAT" />);

    const checkbox = container.querySelector("input");
    fireEvent.click(checkbox);

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
