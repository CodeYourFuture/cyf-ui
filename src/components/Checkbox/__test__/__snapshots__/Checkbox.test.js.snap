// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> test matches snapshot 1`] = `
.c0 {
  color: "yellow";
  background: "blue";
  font-size: 18px;
}

<label
  class="c0"
>
  <input
    checked=""
    type="checkbox"
  />
  include VAT
</label>
`;
